### PR TITLE
DOTNETLIB-5 Issue loading should handle zero-length file level issues

### PR DESCRIPTION
**Pull request by tvsonar**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [SarifParser10.java#L85](https://github.com/SonarSource/sonar-dotnet-shared-library/blob/1563b6501207f0a3488061191d12661a15360787/src/main/java/org/sonarsource/dotnet/shared/sarif/SarifParser10.java#L85): The Cyclomatic Complexity of this method "handleAnalysisTargetElement" is 14 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/org/sonarsource/dotnet/shared/sarif/SarifParser10.java [hunk 1]
```diff
@@ -107,7 +107,11 @@ private static boolean handleAnalysisTargetElement(String ruleId, String message
       int endColumn = endColumnOrNull != null ? endColumnOrNull.getAsInt() : (endLineOrNull != null ? (endLine == startLine ? startColumn : 1) : startColumn);
       int endLineOffset = endColumn - 1;
 
-      callback.onIssue(ruleId, absolutePath, message, startLine, startLineOffset, endLine, endLineOffset);
+      if (startColumn == 1 && endColumn == 1 && startLine == 1 && endLine == 1) {
```
Comment (janos-ss) [inline src/main/java/org/sonarsource/dotnet/shared/sarif/SarifParser10.java]: A minor thing, but isn't the line number generally more discriminant than the column? That is, I assume that there are fewer issues reported about line 1 than column 1. So if you rearrange the conditions to `startLine == 1 && endLine == 1 && startColumn == 1 && endColumn == 1` a `false` conclusion will be reached faster (with fewer evaluated conditions)

File: src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractSensor.java [hunk 2]
```diff
@@ -125,6 +125,27 @@ public void onProjectIssue(String ruleId, String message) {
...
+      }
+
+      InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates().hasAbsolutePath(absolutePath));
+      if (inputFile == null) {
+        return;
+      }
+
+      NewIssue newIssue = context.newIssue();
+      newIssue
+              .forRule(RuleKey.of(repositoryKey, ruleId))
```
Comment (janos-ss) [inline src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractSensor.java]: This indentation doesn't follow the Sonar convention ;-)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [SarifParser10.java#L85](https://github.com/SonarSource/sonar-dotnet-shared-library/blob/bc02de37acaa625c9c74f0457e4122b144dcf43b/src/main/java/org/sonarsource/dotnet/shared/sarif/SarifParser10.java#L85): The Cyclomatic Complexity of this method "handleAnalysisTargetElement" is 14 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Pull request merged**